sql/analyzer/resolve_tables_test.go

```diff
@@ -97,20 +98,24 @@ func TestResolveViews(t *testing.T) {
 
 	f := getRule("resolve_tables")
 
-	table := memory.NewTable("mytable", sql.Schema{{Name: "i", Type: sql.Int32}})
+	table := memory.NewTable("mytable", sql.Schema{{Name: "i", Type: sql.Int32, Source: "mytable"}})
 	db := memory.NewDatabase("mydb")
 	db.AddTable("mytable", table)
 
-	// Resolved plan that corresponds to query "SELECT i FROM mytable"
-	subquery := plan.NewProject(
-		[]sql.Expression{
-			expression.NewGetFieldWithTable(
-				1, sql.Int32, table.Name(), "i", true),
-		},
-		plan.NewResolvedTable(table),
+	subqueryDefinition := plan.NewSubqueryAlias(
+		"myview",
+		plan.NewProject(
+			[]sql.Expression{expression.NewUnresolvedColumn("i")},
+			plan.NewUnresolvedTable("mytable", ""),
+		),
+	)
+	subqueryAnalyzed := plan.NewSubqueryAlias(
+		"myview",
+		plan.NewResolvedTable(table.WithProjection([]string{"i"})),
 	)
-	subqueryAlias := plan.NewSubqueryAlias("myview", subquery)
-	view := sql.NewView("myview", subqueryAlias)
+	expected := fmt.Sprintf("%v", subqueryAnalyzed)
```

Comment:
Would move this down to right above where it's used, and add a comment explaining why this is being used for equality